pandas/core/arrays/timedeltas.py

```diff
@@ -133,15 +133,31 @@ def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
         result._freq = freq
         return result
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE):
+    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
+        verify_integrity = True
 
         freq, freq_infer = dtl.maybe_infer_freq(freq)
 
-        values = np.array(values, copy=False)
-        if values.dtype == np.object_:
-            values = array_to_timedelta64(values)
+        values, inferred_freq = sequence_to_td64ns(values, copy=copy,
+                                                   unit=None)
+        if inferred_freq is not None:
+            if freq is not None and freq != inferred_freq:
+                raise ValueError('Inferred frequency {inferred} from passed '
+                                 'values does not conform to passed frequency '
+                                 '{passed}'
+                                 .format(inferred=inferred_freq,
+                                         passed=freq.freqstr))
+            elif freq_infer:
+                freq = inferred_freq
+                freq_infer = False
+            verify_integrity = False
 
         result = cls._simple_new(values, freq=freq)
+        # check that we are matching freqs
+        if verify_integrity and len(result) > 0:
```

Comment:
can't you use freq_infer rather than verify_integrity here, seems kind of clunky

Reply:
ill double check, but probably.  This is a holdover from the Index behavior, where verify_integrity is also a kwarg for `__new__` (which probably should be deprecated)

Reply:
Turns out we can't use freq_infer, but we can use `inferred_freq is None`